packages/location_permissions/android/src/main/java/com/baseflow/location_permissions/LocationPermissionsPlugin.java

```diff
@@ -42,13 +42,15 @@
   private static final int PERMISSION_STATUS_DENIED = 1;
   private static final int PERMISSION_STATUS_GRANTED = 2;
   private static final int PERMISSION_STATUS_RESTRICTED = 3;
+  private static final int PERMISSION_STATUS_WHEN_IN_USE = 4;
 
   @Retention(RetentionPolicy.SOURCE)
   @IntDef({
-    PERMISSION_STATUS_UNKNOWN,
-    PERMISSION_STATUS_DENIED,
-    PERMISSION_STATUS_GRANTED,
-    PERMISSION_STATUS_RESTRICTED,
+          PERMISSION_STATUS_UNKNOWN,
+          PERMISSION_STATUS_DENIED,
+          PERMISSION_STATUS_GRANTED,
+          PERMISSION_STATUS_RESTRICTED,
+          PERMISSION_STATUS_WHEN_IN_USE,
```

Comment:
I think we should not change the `PERMISSION_STATUS` values. Depending on which permissions are requested we should simply return if permissions are granted or denied (or unknown in case the service is not supported). 

When the user requests permissions for `locationAlways` the code should do the following:
- If running on Android Q and above and the "android.permission.ACCESS_BACKGROUND_LOCATION" permission exists in the AndroidManifest, check permissions for the Background Location Permission;
- else fallback to current implementation.